plugins/in_emitter/emitter.c

```diff
@@ -237,6 +237,9 @@ static int in_emitter_ingest_ring_buffer(struct flb_input_instance *in,
         flb_sds_destroy(ec.tag);
         msgpack_sbuffer_destroy(&ec.mp_sbuf);
     }
+    if (ret < 0) {
+        return -1;
+    }
```

Comment:
Why the two conditions, why not return in the previous < 0?

Reply:
I think I was trying to avoid problems where the buffer would not be drained if there was a single append that caused an error.

Reply:
I've been looking at this and pondering what to do about it. This code could wind up dropping ring buffers if `flb_log_append` ever throws an error as it is. Fixing this seems out of scope for this PR but we should tackle it. I actually started working on a solution that uses `lwrb_peek` and `lwrb_skip` to leave the ring buffer untouched when `flb_log_append` returns an error.

Reply:
I am going to revert the change in this branch and open another one which will actually leave buffers in the ring buffer if `flb_input_log_append` returns an error. This is enough of a fundamental change I believe it warrants it's own PR.

Reply:
@niedbalski any chance of getting another look at this?